idf_build_apps/app.py

```diff
@@ -184,6 +188,19 @@ def __init__(
 
         self._sdkconfig_files, self._sdkconfig_files_defined_target = self._process_sdkconfig_files()
 
+    @classmethod
+    def from_another(cls, other: 'App', **kwargs) -> 'App':
+        """Init New App from another, with different parameters.
+
+        eg: new_app = [Cmake]App.from_another(app, config='debug')
+        """
+        n_kwargs = copy.deepcopy(other._kwargs)
+        n_kwargs.update(kwargs)
+        # positional args
+        app_dir = n_kwargs.pop('app_dir')
+        target = n_kwargs.pop('target')
+        return cls(app_dir, target, **n_kwargs)
```

Comment:
`app_dir` and `target` are not positional arguments.

You can pass them as kwargs as well.

```suggestion
        return cls(**n_kwargs)
```

Reply:
they are positional args:

```
    def __init__(
        self,
        app_dir: str,
        target: str,
        *,
        work_dir: t.Optional[str] = None,
        build_dir: str = 'build',
        build_log_filename: t.Optional[str] = None,
        size_json_filename: t.Optional[str] = None,
        **kwargs: t.Any,
    ) -> None:
```

Reply:
```python
def foo(a, b, /, c, d, *, e, f):
    pass
```

a, b are positional args,
c, d are normal ones,
e, f are keyword args.

also explained here: https://github.com/espressif/idf-build-apps/blob/main/docs/migration/1.x_to_2.x.md#normal-arguments-to-keyword-only-arguments